app/main.py

```diff
@@ -1,25 +1,78 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self,
+                     owner: SlideLimitationValidator,
+                     name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self,
+                instance: SlideLimitationValidator,
+                owner: SlideLimitationValidator,
+                ) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self,
+                instance: SlideLimitationValidator,
+                value: SlideLimitationValidator,
+                ) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"Value should not be "
+                             f"less than {self.min_amount} "
+                             f"and greater than "
+                             f"{self.max_amount}")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(self,
+                 name: str,
+                 limitation_class: type[SlideLimitationValidator],
```

Comment:
use single typehint style
```suggestion
                 limitation_class: SlideLimitationValidator,
```